zerver/views/presence.py

```diff
@@ -48,6 +49,11 @@ def get_presence_backend(
     if target.is_bot:
         raise JsonableError(_("Presence is not supported for bot users."))
 
+    if settings.LIMIT_PRESENCE_VISIBILITY_FOR_GUESTS and not check_can_access_user(
```

Comment:
I don't think we need to take care of loading time here as this endpoint is not used by webapp as of now, but just added the condition to check server level setting for consistency.

Reply:
Agreed on both points that this makes sense.